website/docs/feature-flag-tutorials/dotnet/implementing-feature-flags-dotnet.md

```diff
@@ -0,0 +1,273 @@
+public record RickAndMortyGraphQLResponse(Data? Data);
+public record Data(Characters? Characters);
+public record Characters(Character[]? Results);
+public record Character
+{
+    public string? Name { get; init; }
+}
+
+```
+
+Run the code:
+
+```sh
+dotnet run
+```
+
+or, run the watcher to make your project automatically reload on file changes.
+
+```sh
+dotnet watch --hotreload run
+```
+
+You should see `There are 20 characters` in your terminal.
+
+## 4. Add the GraphQL endpoint
+
+The point of this tutorial is to mimic a real-world scenario where you would migrate from a REST API to GraphQL and rely on a boolean feature flag to toggle between the versions. So far, we’ve just used REST. Now, let’s add the GraphQL version. The GraphQL endpoint is `https://rickandmortyapi.com/graphql`.
+
+Let’s create a static feature flag, for now, just to test that we can call both versions successfully.
+
+```csharp
+app.MapGet("/", async (HttpClient httpClient) =>
+{
+        bool isGraphql = true; // Change to true to use GraphQL
+
+        if (isGraphql)
+        {
+            var query = new {
+                query = @"
+                {
+                    characters {
+                        results {
+                            name
+                        }
+                    }
+                }
+                "
+            };
+
+            var response = await httpClient.PostAsJsonAsync("https://rickandmortyapi.com/graphql", query);
+            var result = await response.Content.ReadFromJsonAsync<RickAndMortyGraphQLResponse>();
+            var characters = result?.Data?.Characters?.Results;
+
+            Console.WriteLine("Hello GraphQL");
+            Console.WriteLine($"There are {characters?.Length} characters");
+            return characters;
+        }
+        else
+        {
+            var response = await httpClient.GetFromJsonAsync<RickAndMortyApiResponse>("https://rickandmortyapi.com/api/character/");
+            var characters = response?.Results;
+
+            Console.WriteLine("Hello REST API");
+            Console.WriteLine($"There are {characters?.Length} characters");
+            return characters;
+        }
+
+
+});
+
+app.Run();
+```
+
+Run the code again:
+
+```sh
+dotnet run
+```
+
+You should see `Hello GraphQL`, followed by `There are 20 characters` in your terminal.
+
+Feel free to change the `isGraphql` variable to try it out.
+
+## 5. Add Unleash to your .NET app
+
+Now, let’s connect our project to Unleash so that you can toggle that feature flag at runtime. If you wanted to, you could also do a gradual rollout, use it for A/B testing, etc.
+
+You’ll need 3 things:
+
+- The Unleash SDK installed
+- The URL of your Unleash instance’s API. It’s `[http://localhost:4242/api/](http://localhost:4242/api/)` for your local version. You’ll want to replace it with your remote instance.
+- The API token we created on our Unleash instance, feel free to create another one if you can’t find it.
```

Comment:
I think you left out the create token part in the Unleash setup above? You do link to the docs further down, but it looks like you intended to guide the user through how to create a token :)